src/services/TestResultsService.ts

```diff
@@ -128,38 +139,83 @@ export class TestResultsService {
   public updateTestResult(systemNumber: string, payload: ITestResult, msUserDetails: IMsUserDetails) {
     this.removeNonEditableAttributes(payload);
     let validationSchema = this.getValidationSchema(payload.vehicleType, payload.testStatus);
-    validationSchema = validationSchema!.keys({countryOfRegistration: Joi.string().valid(countryOfRegistration).required().allow("", null)});
-    validationSchema = validationSchema!.optionalKeys(["testStationType", "testerEmailAddress", "testEndTimestamp", "systemNumber", "vin"]);
+    const testTypesValidationErrors = this.validateTestTypes(payload);
+    if (testTypesValidationErrors.length) {
+      return Promise.reject(new HTTPError(400, {errors: testTypesValidationErrors}));
+    }
+    // temporarily remove testTypes to validate only vehicle details and append testTypes to the payload again after the validation
+    const {testTypes} = payload;
+    delete payload.testTypes;
```

Comment:
Need to understand why the testTypes are deleted?

Reply:
they are not actually deleted, they are removed from the payload temporarily and then added back again. I did this because we are already validating each testType individually (and defects as well, since they are at testType level). For the rest of the test-result object (vehicle details) I'm reusing the validation schemas implemented by VTA, but these validation schemas already contain testTypes validation, which are very general. We don't want to validate them again, since we already did previously. So, I am removing the testTypes from the payload, I'm extending the validation schema and specify that testTypes shouldn't be validated and if the validation is successful I'm appending the validated testTypes again to the validated payload.